### PR TITLE
feat(UBA): :zap: optimize uba fee calculation's spoke pool clients

### PR DESCRIPTION
**Pull request by james-a-morris**
This change adds the concept of a UBAFeeSpokeStore which stores information needed for one half of the computation values of a UBAFee Calculation. Two of these stores can be combined using the UBAFeeCalculator wrapper to facilitate the calculation of a valid UBA Fee.

Breaking Change: The class definition of the UBAFeeCalculator has changed.

### CODE REVIEW

File: src/UBAFeeCalculator/UBAFeeSpokeCalculator.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
+import { BigNumber } from "ethers";
+import { UBAFlowRange, UbaFlow, isUbaInflow, isUbaOutflow } from "../interfaces";
+import { toBN } from "../utils";
+import UBAConfig from "./UBAFeeConfig";
+import { getDepositBalancingFee, getRefundBalancingFee } from "./UBAFeeUtility";
+
+/**
+ * @file UBAFeeSpokeCalculator.ts
+ * @description UBA Fee Spoke Store - meant to memoize the running balance of a spoke so that we don't have to recompute it N^2 times in the UBA Fee Calculator
```
Comment (amateima) [inline src/UBAFeeCalculator/UBAFeeSpokeCalculator.ts:9]: So "Store" it's about the `cachedRunningBalance`  variable?

File: src/UBAFeeCalculator/UBAFeeSpokeCalculator.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+  /**
+   * Clears the cached running balance of the spoke
+   */
+  public clearCachedRunningBalance(): void {
+    this.cachedRunningBalance = {};
+  }
+
+  /**
+   *
+   */
```
Comment (dohaki) [inline src/UBAFeeCalculator/UBAFeeSpokeCalculator.ts:102]: nit: this will follow in a separate PR?

**Review by dohaki — APPROVED** [review]
LGTM 👍 Left one nit

File: src/UBAFeeCalculator/UBAFeeSpokeCalculator.ts (hunk 2 above)
Comment (james-a-morris) [reply src/UBAFeeCalculator/UBAFeeSpokeCalculator.ts]: Done! Great callout

**Pull request merged**